sources/src/include/commpipe.h

```diff
@@ -48,6 +48,11 @@ STATIC_INLINE void destroy_comm_pipe (smp_comm_pipe *p)
     uae_sem_destroy (&p->lock);
     uae_sem_destroy (&p->reader_wait);
     uae_sem_destroy (&p->writer_wait);
+    if (p->data)
+    {
+		free(p->data);
+		p->data = NULL;
+    }
```

Comment:
It's not necessary to check for null. `free(NULL)` is always a no-op. This can just be:

    free(p->data);
    p->data = NULL;

Reply:
Ah, I come from a time when you did have to check that  :)

I do like the clarity of checking for NULL. I know it's a matter of opinion...